tests/CloudEventFunctionsWrapperTest.php

```diff
@@ -0,0 +1,102 @@
+ */
+class CloudEventFunctionWrapperTest extends TestCase
+{
+    private static $functionCalled = false;
+
+    public function testInvalidRequestBody()
+    {
+        $this->expectException('RuntimeException');
+        $this->expectExceptionMessage('Could not parse CloudEvent: Syntax error');
+        $request = new ServerRequest('POST', '/', [], 'notjson');
+        $cloudEventFunctionWrapper = new CloudEventFunctionWrapper([$this, 'invokeThis']);
+        $cloudEventFunctionWrapper->execute($request);
+    }
+
+    public function testWithFullCloudEvent()
+    {
+        $CloudEventFunctionWrapper = new CloudEventFunctionWrapper([$this, 'invokeThis']);
+        $request = new ServerRequest('POST', '/', [
+            'ce-id' => '1413058901901494',
+            'ce-source' => '//pubsub.googleapis.com/projects/MY-PROJECT/topics/MY-TOPIC',
+            'ce-specversion' => '1.0',
+            'ce-type' => 'com.google.cloud.pubsub.topic.publish',
+            'ce-datacontenttype' => 'application/json',
+            'ce-dataschema' => 'type.googleapis.com/google.logging.v2.LogEntry',
+            'ce-subject' => 'My Subject',
+            'ce-time' => '2020-12-08T20:03:19.162Z',
+        ], json_encode([
+            "message" => [
+                "data" => "SGVsbG8gdGhlcmU=",
+                "messageId" => "1408577928008405",
+                "publishTime" => "2020-08-06T22:31:14.536Z"
+            ],
+            "subscription" => "projects/MY-PROJECT/subscriptions/MY-SUB"
+        ]));
+        $CloudEventFunctionWrapper->execute($request);
+        $this->assertTrue(self::$functionCalled);
+    }
+
+    public function invokeThis(CloudEvent $cloudevent)
+    {
+        self::$functionCalled = true;
+        $this->assertEquals('1413058901901494', $cloudevent->getId());
+        $this->assertEquals('//pubsub.googleapis.com/projects/MY-PROJECT/topics/MY-TOPIC', $cloudevent->getSource());
+        $this->assertEquals('1.0', $cloudevent->getSpecVersion());
+        $this->assertEquals('com.google.cloud.pubsub.topic.publish', $cloudevent->getType());
+        $this->assertEquals('application/json', $cloudevent->getDataContentType());
+        $this->assertEquals('type.googleapis.com/google.logging.v2.LogEntry', $cloudevent->getDataSchema());
+        $this->assertEquals('My Subject', $cloudevent->getSubject());
+        $this->assertEquals('2020-12-08T20:03:19.162Z', $cloudevent->getTime());
+    }
+    
+    public function testWithNotFullButValidCloudEvent()
+    {
+        $CloudEventFunctionWrapper = new CloudEventFunctionWrapper([$this, 'invokeThisPartial']);
+        $request = new ServerRequest('POST', '/', [
+            'ce-id' => 'fooBar',
+            'ce-source' => 'my-source',
+            'ce-specversion' => '1.0',
+            'ce-type' => 'my.type',
+        ], json_encode([
+            "key" => "value"
+        ]));
+        $CloudEventFunctionWrapper->execute($request);
+        $this->assertTrue(self::$functionCalled);
```

Comment:
These methods should set `self:$functionCalled = false;` at the beginning to ensure the static value isn't propagated across tests (as is currently happening)

Reply:
Done.